lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp

```diff
@@ -737,15 +752,13 @@ struct FoldBroadcastToFMAOp : public OpConversionPattern<aievec::FMAOp> {
 };
 
 struct ConvertMulAddToAIEVecFMAOpPattern
-    : public OpConversionPattern<arith::AddIOp> {
-  using OpConversionPattern<arith::AddIOp>::OpConversionPattern;
+    : public OpConversionPattern<aievec::AddOp> {
```

Comment:
Should we move this to AIEVecOptimizations.cpp?

Reply:
It's an option, yes, but for v1 there's an extra wrinkle to sort out first. The reason we do MUL+ADD -> MAC in the dialect conversion is because the way we convert broadcast is by merging it with FMA. If we leave aievec.mul and aievec.add with the intention of fusing them together into a FMA in the optimization layer (which is very reasonable, and my preference), then we leave vector.broadcast alive after conversion. So, the order of operations would be to first deal with the lowering of broadcast for v1, and then move everything to its rightful place.

Good eye, though, you're very quickly catching up with all the technical debt we've left behind :-)